probe/pkg/probe/probe_test.go

```diff
@@ -0,0 +1,402 @@
+}
+
+func TestCreateCentral(t *testing.T) {
+	tt := []struct {
+		testName     string
+		wantErr      bool
+		errContains  string
+		mockFMClient *fleetmanager.PublicClientMock
+	}{
+		{
+			testName: "create central happy path",
+			wantErr:  false,
+			mockFMClient: &fleetmanager.PublicClientMock{
+				CreateCentralFunc: func(ctx context.Context, async bool, request public.CentralRequestPayload) (public.CentralRequest, *http.Response, error) {
+					central := public.CentralRequest{
+						Status:       constants.CentralRequestStatusAccepted.String(),
+						InstanceType: types.STANDARD.String(),
+					}
+					return central, nil, nil
+				},
+				GetCentralByIdFunc: func(ctx context.Context, id string) (public.CentralRequest, *http.Response, error) {
+					central := public.CentralRequest{
+						Status:       constants.CentralRequestStatusReady.String(),
+						InstanceType: types.STANDARD.String(),
+					}
+					return central, nil, nil
+				},
+			},
+		},
+		{
+			testName:    "create central fails on internal server error",
+			wantErr:     true,
+			errContains: "creation of central instance failed",
+			mockFMClient: &fleetmanager.PublicClientMock{
+				CreateCentralFunc: func(ctx context.Context, async bool, request public.CentralRequestPayload) (public.CentralRequest, *http.Response, error) {
+					central := public.CentralRequest{}
+					err := errors.Errorf("%d", http.StatusInternalServerError)
+					return central, nil, err
+				},
+			},
+		},
+		{
+			testName:    "central not ready on internal server error",
+			wantErr:     true,
+			errContains: "central instance id-42 did not reach ready state",
+			mockFMClient: &fleetmanager.PublicClientMock{
+				CreateCentralFunc: func(ctx context.Context, async bool, request public.CentralRequestPayload) (public.CentralRequest, *http.Response, error) {
+					central := public.CentralRequest{
+						Id:           "id-42",
+						Name:         "probe-pod-42",
+						Status:       constants.CentralRequestStatusAccepted.String(),
+						InstanceType: types.STANDARD.String(),
+					}
+					return central, nil, nil
+				},
+				GetCentralByIdFunc: func(ctx context.Context, id string) (public.CentralRequest, *http.Response, error) {
+					central := public.CentralRequest{}
+					err := errors.Errorf("%d", http.StatusInternalServerError)
+					return central, nil, err
+				},
+			},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.testName, func(t *testing.T) {
+			probe, err := New(testConfig, tc.mockFMClient, nil)
+			require.NoError(t, err, "failed to create probe")
+			ctx, cancel := context.WithTimeout(context.TODO(), testConfig.ProbeRunTimeout)
+			defer cancel()
+
+			central, err := probe.createCentral(ctx)
+
+			if tc.wantErr {
+				assert.ErrorContains(t, err, tc.errContains, "expected an error during probe run")
+			} else {
+				assert.Equal(t, constants.CentralRequestStatusReady.String(), central.Status, "central not ready")
+				assert.NoError(t, err, "failed to create central")
```

Comment:
You probably want those to be switched and a `required`. If you were to receive an unexpected error in `probe.CreateCentral`, you would get `central == nil` which would lead to a panic when accessing `central.Status`.
```suggestion
				require.NoError(t, err, "failed to create central")
				assert.Equal(t, constants.CentralRequestStatusReady.String(), central.Status, "central not ready")

```